samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -587,8 +628,13 @@ static void gen_onpowerup_set(struct bt_mesh_model *model,
 			      struct bt_mesh_msg_ctx *ctx,
 			      struct net_buf_simple *buf)
 {
+	struct generic_onpowerup_state *state = model->user_data;
+
 	gen_onpowerup_set_unack(model, ctx, buf);
-	gen_onpowerup_get(model, ctx, buf);
+
+	if (state->val <= 0x02) {
+		gen_onpowerup_get(model, ctx, buf);
```

Comment:
You're still using a state struct variable instead of a stack variable here. Also, isn't this broken since `set_unack()` will change the parsing state of the net_buf (with its `net_buf_pull_*()` calls), so what you're giving to `gen_powerup_get()` will not be what it expects (in fact it'll likely result in reading invalid memory). Did you actually test this?

Reply:
Done !!